### PR TITLE
Add pyright to check python types

### PR DESCRIPTION
**Pull request by gregtatum**
This starts some incremental static type checking for the project in the utils/* directory.

### CODE REVIEW

File: taskcluster/kinds/tests/kind.yml [hunk 1]
```diff
@@ -91,6 +91,20 @@ tasks:
             task lint-black
     run-on-tasks-for: ["github-push", "github-pull-request"]
 
+  lint-pylance:
+    # Run python's pylance type checker.
```
Comment (nordzilla) [inline taskcluster/kinds/tests/kind.yml:95]: > Question

I think maybe this is supposed to be `pright` instead of `pylance`. Is that right?

**Review by nordzilla — APPROVED** [review]
Changes all look reasonable to me, though I'm not a Python expert. 

I am certainly in favor of having type checkers whenever possible.

I just have the one inline comment question about `pylance` vs. `pyright`

> Edit: Looks like the pylance thing failed CI as is, so I think it was just an oversight.

File: taskcluster/kinds/tests/kind.yml (hunk 1 above)
Comment (gregtatum) [reply taskcluster/kinds/tests/kind.yml:95]: Whoops, yeah pylance is the name of the VSCode extension. That's a mistake.

**Comment by gregtatum** [comment]
Looks like it requires the world to be installed to find all of the modules, and is currently failing.

**Pull request merged**